btcd.go

```diff
@@ -88,6 +88,17 @@ func btcdMain(serverChan chan<- *server) error {
 		defer pprof.StopCPUProfile()
 	}
 
+	// Write mem profile if requested.
+	if cfg.MemoryProfile != "" {
+		f, err := os.Create(cfg.MemoryProfile)
+		if err != nil {
+			btcdLog.Errorf("Unable to create memory profile: %v", err)
+			return err
+		}
+		defer f.Close()
+		defer pprof.WriteHeapProfile(f)
+	}
```

Comment:
could add a `defer runtime.GC()` call at the end so that the heap profile is accurate as far as what's actually lingering

Reply:
@Roasbeef any opinions on this? It'll give us better data but it may slow things down during shutdown for normal users.

Reply:
IIUC, that would only be active for users already running with a flag that strictly reserved for profiling. So I think that's fine.

Reply:
Added a runtime.GC() so that garbage collection happens before writes